build/changelog.py

```diff
@@ -53,23 +89,30 @@ def parse_args():
     parser = argparse.ArgumentParser()
     parser.add_argument(
         "--repo_url", default="https://github.com/open-policy-agent/opa")
+    parser.add_argument("token", nargs="?", 
+                        default="", help="GitHub API token")
     parser.add_argument("from_version", nargs="?",
                         default=get_latest_tag(), help="start of changes")
     parser.add_argument("to_commit", nargs="?",
                         default="HEAD", help="end of changes")
     return parser.parse_args()
 
 
+maintainers = get_maintainers()
+
 def main():
 
     args = parse_args()
+    author = ""
     changelog = {}
 
     for commit_id in get_commit_ids(args.from_version, args.to_commit):
         commit_message = get_commit_message(commit_id)
-        group, line = get_changelog_message(commit_message, args.repo_url)
+        if not is_maintainer(commit_message):
+            author = get_github_id(commit_id, args.token)
```

Comment:
It doesn't look like the GitHub username is being cached as described--this means we're fetching the username repeatedly for the same external contributor. I think we should implement the caching to avoid the rate limits. For example, this would remove 50% of the fetches we do for v0.19.0..v0.20.0.

Reply:
Done!